File: 0728-self-dividing-numbers/0728-self-dividing-numbers.cpp
class Solution {
public:
    bool f(int x){
        int n=x;
        while(n!=0){
            int y=n%10;
            if(y!=0 && x%y==0){
                n/=10;
                continue;
            }
            else return false;
        }
        return true;
    }
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int>ans;

        for(int i=left;i<=right;i++){
            int x=i;
            if(f(i))ans.push_back(x);
        }
        return ans;
    }
};

File: 0728-self-dividing-numbers/0728-self-dividing-numbers.py
class Solution(object):
    def f(self,x,n):
        if(n==0):
            return True

        y=n%10
        if(y==0 or x%y!=0):
            return False

        return self.f(x,n//10)

    def selfDividingNumbers(self, left, right):
        ans=[]

        for i in range(left,right+1):
            x=i
            if(self.f(x,i)):
                ans.append(x)

        return ans
        